cpmpy/expressions/globalconstraints.py

```diff
@@ -582,3 +585,65 @@ def deepcopy(self, memodict={}):
         return Cumulative(*copied_args)
 
 
+class GlobalCardinalityCount(GlobalConstraint):
+    """
+        GlobalCardinalityCount(a,gcc): Collect the number of occurrences of each value 0..a.ub in gcc.
```

Comment:
Can `a` contain arbitrary expressions, not only pure values? I guess yes, right?

Reply:
with the bounds calculations pr that would be possible, as is only values or intvars are allowed